test/e2e/storage/drivers/csi.go

```diff
@@ -61,35 +62,47 @@ type hostpathCSIDriver struct {
 }
 
 func initHostPathCSIDriver(name string, config testsuites.TestConfig, manifests ...string) testsuites.TestDriver {
-	return &hostpathCSIDriver{
+	driver := &hostpathCSIDriver{
 		driverInfo: testsuites.DriverInfo{
 			Name:        name,
 			FeatureTag:  "",
 			MaxFileSize: testpatterns.FileSizeMedium,
 			SupportedFsType: sets.NewString(
 				"", // Default fsType
 			),
-			Capabilities: map[testsuites.Capability]bool{
-				testsuites.CapPersistence: true,
-			},
 
 			Config: config,
 		},
 		manifests: manifests,
 	}
+
+	if name == "csi-hostpath-v0" {
+		driver.driverInfo.Capabilities = map[testsuites.Capability]bool{
+			testsuites.CapPersistence: true,
+		}
+	} else {
+		driver.driverInfo.Capabilities = map[testsuites.Capability]bool{
+			testsuites.CapPersistence: true,
+			testsuites.CapDataSource:  true,
+		}
+	}
+	return driver
```

Comment:
I find it cleaner to not set Capabilities at all in `initHostPathCSIDriver`.

Instead do that in `InitHostPathCSIDriver` and `InitHostPathV0CSIDriver`, for example:
```
func InitHostPathCSIDriver(config testsuites.TestConfig) testsuites.TestDriver {
    driver := initHostPathCSIDriver("csi-hostpath", config, ...)
    driver.driverInfo.Capabilities = map[testsuites.Capability]bool{
			testsuites.CapPersistence: true,
			testsuites.CapDataSource:  true,
		}
    return driver
}
```

Reply:
That, or add another parameter to `initHostPathCSIDriver`.

Reply:
Updated by add parameter to `initHostPathCSIDriver`